### PR TITLE
request-handler: emit error when failed to parse arg2/arg3

### PR DESCRIPTION
**Pull request by Raynos**
We completely swallowed errors when failing to parse args.

This is BAD (tm).

r: @jcorbin @rf @kriskowal

### CODE REVIEW

**Comment by Matt-Esch** [comment]
:+1:

File: request-handler.js [hunk 1]
```diff
@@ -54,8 +54,19 @@ RequestCallbackHandler.prototype.handleRequest = function handleRequest(req, bui
     var res;
     if (req.streamed) {
         req.withArg23(function onArg23(err, arg2, arg3) {
+            if (err) {
+                req.channel.logger.warn(
+                    'Could not parse arg3 for streaming inreq',
```
Comment (jcorbin) [inline request-handler.js]: or arg2, you don't necessarily know that the problem was with arg3

File: request-handler.js [hunk 2]
```diff
@@ -54,8 +54,19 @@ RequestCallbackHandler.prototype.handleRequest = function handleRequest(req, bui
...
+                    req.extendLogInfo({
+                        error: err
+                    })
+                );
+
+                return req.emitError(err);
```
Comment (jcorbin) [inline request-handler.js]: won't it log about it? shouldn't we just wrap it here in a `arg23somethingerorr'? maybe `req#withArg23` should do such wrapping for us?

Comment (Raynos) [inline request-handler.js]: I dont trust it; in request might have already errored and just drop the second error on the floor.

**Comment by jcorbin** [comment]
:ok: once pass tests

**Pull request merged**